packages/webapp/src/components/Icons/index.jsx

```diff
@@ -1,3 +1,37 @@
+/*
+ *  Copyright 2024 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <<https://www.gnu.org/licenses/>.>
+ */
+
 import Cross from './cross';
+import styles from './styles.module.scss';
+import clsx from 'clsx';
+import iconMap from './icons';
 
 export { Cross };
+
+export { iconMap };
+
+const Icon = ({ iconName, circle = false, className = '', ...rest }) => {
+  const Icon = iconMap[iconName];
+
+  return (
+    <div className={styles.displayBlock}>
```

Comment:
Right, what was I seeing? lol . Let me write this out and see if it makes sense.

...in `TransactionItem`, `IconListItem` and `IconTile` there was a wrapper div (now removed in this PR) used for positioning items like checkbox, text, icon using flex. When there was not a container div around icon, the icons would not align as expected because they would inherit I think the flex-item and not find center properly.

To your point I am not seeing that behaviour if I remove this wrapper. But I think I did that as a way to _actually prevent_ overwriting the display property using className. Since we should 99.9% always want to position the Icon as if it were "one piece".

Yeah so I think this wrapper is distinct that it remain uneditable for now. And that if someone were to need a new type of icon that inherited display they could expand this component to be like `<List /> `or` <Tiles />` with multiple `<Icon />` kinds.

Does that make any sense? haha

Reply:
I think I got it!